src/lipizzaner_master.py

```diff
@@ -157,85 +162,112 @@ def _terminate(self, stop_clients=True, return_code=-1):
-                    filename = '{}{}.pkl'.format(GENERATOR_PREFIX, source)
-                    torch.save(generator.genome.net.state_dict(),
-                               os.path.join(output_dir, 'generator-{}.pkl'.format(source)))
+                    source = generator.source.replace(":", "-")
+                    filename = "{}{}.pkl".format(GENERATOR_PREFIX, source)
+                    torch.save(
+                        generator.genome.net.state_dict(), os.path.join(output_dir, "generator-{}.pkl".format(source)),
+                    )
 
-                    with open(os.path.join(output_dir, 'mixture.yml'), "a") as file:
-                        file.write('{}: {}\n'.format(filename, weights_generator[generator.source]))
+                    with open(os.path.join(output_dir, "mixture.yml"), "a") as file:
+                        file.write("{}: {}\n".format(filename, weights_generator[generator.source]))
 
                 for discriminator in discriminator_pop.individuals:
-                    source = discriminator.source.replace(':', '-')
-                    filename = '{}{}.pkl'.format(DISCRIMINATOR_PREFIX, source)
-                    torch.save(discriminator.genome.net.state_dict(),
-                               os.path.join(output_dir, filename))
+                    source = discriminator.source.replace(":", "-")
+                    filename = "{}{}.pkl".format(DISCRIMINATOR_PREFIX, source)
+                    torch.save(
+                        discriminator.genome.net.state_dict(), os.path.join(output_dir, filename),
+                    )
+                    if "ssgan" in self.cc.settings["network"]["name"]:
+                        filename = "{}{}_classification_layer.pkl".format(DISCRIMINATOR_PREFIX, source)
+                        torch.save(
+                            discriminator.genome.classification_layer.state_dict(), os.path.join(output_dir, filename),
+                        )
 
                 # Save images
-                dataset = MixedGeneratorDataset(generator_pop,
-                                                weights_generator,
-                                                self.cc.settings['master']['score_sample_size'],
-                                                self.cc.settings['trainer']['mixture_generator_samples_mode'])
-                image_paths = self.save_samples(dataset, output_dir, dataloader)
-                self._logger.info('Saved mixture result images of client {} to target directory {}.'
-                                  .format(node_name, output_dir))
+                dataset = MixedGeneratorDataset(
+                    generator_pop,
+                    weights_generator,
+                    self.cc.settings["master"]["score_sample_size"],
+                    self.cc.settings["trainer"]["mixture_generator_samples_mode"],
+                )
+
+                if "gaussian_" in self.cc.settings["dataloader"]["dataset_name"]:
```

Comment:
Can these be handle by default arguments to avoid controlflow here

Reply:
It is already handled with default arguments.